remotespark/livyclientlib/livysession.py

```diff
@@ -68,16 +68,20 @@ def start(self):
 
         self.logger.debug("Session '{}' started.".format(self.kind))
 
-    def create_sql_context(self):
+    def create_sql_context(self, msg_printer):
         """Create a sqlContext object on the session. Object will be accessible via variable 'sqlContext'."""
         if self.started_sql_context:
             return
 
         self.logger.debug("Starting '{}' sql and hive session.".format(self.kind))
 
+        msg_printer.print_message('Creating SQL Context...')
```

Comment:
If we implemented this, I would definitely want the dots to be dismissed after we're done waiting for idle, since they're just noise.

It looks like we're only calling `wait_for_idle` to wait before creating the SQL context; we never actually call that method before cell executions.  We could add another call to this method before cell executions if you wanted to implement this.

I'm not sure if I like the idea all that much myself.  Usually dots that programs pop up suggest the program is busy doing some useful work, when the opposite is true here.  It might be confusing for users if they sporadically get 1 or 2 dots, and then get no more dots for another 10 minutes before their result is displayed. 

What is it that you would want to communicate with output text like this?  I think it might be better to identify those scenarios, recognize when they happen, and pop up with disposable text messages that capture what's happening for the user (e.g. "This query is taking longer than usual", etc.)